### DISCUSSION TITLE
Capstone ML - Random Forest Predict Error Cyber Bully

### DISCUSSION THREAD

**Discussion by angeliainge** [teamalgoritma/community#345] [Q&A] [answered]
Halo tim algoritma,

saya ingin menanyakan perihal error saat predict model saya ke data test dgn metode randomforest

<img width="498" alt="Screen Shot 2022-09-25 at 1 18 58 AM" src="https://user-images.githubusercontent.com/107099548/192112930-e371326a-9c8b-41b8-ab92-4a5a800175d0.png">
Untuk variable yang missing jujur saja saya juga bingung, karena saya menggunakan label yg sama untuk predict di Naive Bayes dan RF (ke data validasi).

Saya sudah coba googling dan yang saya tangkapini terjadi karena ada perbedaan leveling (di train dan validation ada yes no, tapi di test NA semua)?

Apakah ada panduan bagaimana yg harus dilakukan bila terjadi spt ini? Karena untuk predict dgn Naive Bayes di test data tidak ada masalah sama sekali

thank you

**Comment by fafilia** [comment, marked as answer]
Dear @angeliainge ,

Boleh ditunjukkan code yang digunakan untuk membuat model rf dan bagaimana hasil output dari pemanggilan data train dan data df_test_bn_rf untuk 5 data teratas?

**Reply by angeliainge** [reply]
Hi kak

terlampir ya, thank you
<img width="300" alt="Screen Shot 2022-09-26 at 11 39 24 AM" src="https://user-images.githubusercontent.com/107099548/192196178-229a4526-41d7-44f2-8652-a75adc35ce5e.png">
<img width="1335" alt="Screen Shot 2022-09-26 at 11 44 25 AM" src="https://user-images.githubusercontent.com/107099548/192196182-413f9ae8-c5d5-4fac-805e-2ed9a84488fd.png">
<img width="494" alt="Screen Shot 2022-09-26 at 11 50 38 AM" src="https://user-images.githubusercontent.com/107099548/192196188-d2a13e18-e2fc-4899-96ec-0fc8f5f10953.png">
<img width="1261" alt="Screen Shot 2022-09-26 at 11 50 43 AM" src="https://user-images.githubusercontent.com/107099548/192196192-388dcb0b-f83d-4769-89ef-91b726f1cb22.png">

<img width="476" alt="Screen Shot 2022-09-26 at 1 17 12 PM" src="https://user-images.githubusercontent.com/107099548/192206211-d10e2afd-5dcf-4ef3-8356-abc5e000fa40.png">

**Reply by fafilia** [reply]
Baik terima kasih untuk jawaban yang dilampirkan. Mungkin bisa dicoba kembali Bu untuk predictnya. Disini untuk melakukan predict, random forest hanya menerima 2 tipe saja yaitu prob untuk menmpilkan probability dan raw untuk menampilkan hasil prediksi berupa label kelas. Oleh karena itu, silahkan di code ibu yang semula type="class" diubah menjadi type="raw".

**Reply by angeliainge** [reply]
hi kak,

masih tidak bisa ya.. 
<img width="482" alt="Screen Shot 2022-09-26 at 2 37 16 PM" src="https://user-images.githubusercontent.com/107099548/192220022-9ba4fbfb-bdc3-4cc0-9ff3-da0a2dfaf9d4.png">
<img width="432" alt="Screen Shot 2022-09-26 at 2 37 26 PM" src="https://user-images.githubusercontent.com/107099548/192220034-9b8f0742-6677-42a2-afc9-87fc139996fb.png">
<img width="473" alt="Screen Shot 2022-09-26 at 2 37 36 PM" src="https://user-images.githubusercontent.com/107099548/192220037-4e5e41c7-8bd3-476a-9636-bcc53cd7c3d1.png">

untuk data train ke validation saya menggunakan "class" tetap bs kak.. apa kah ada pengaruh karena saya pakai Mac?

**Reply by fafilia** [reply]
Boleh dipastikan terlebih dahulu Bu, untuk data test memang semua datanya pada label bully NA semua karena kolom tersebut yang nantinya kita isi dengan data hasil prediksi. Nah masalahnya adalah ketika ibu akan melakukan prediksi ke data test, pastikan semua proses cleansing dan penyesuaian yang dilakukan untuk data train dan data validasi juga diterapkan sama persis pada data test (bentuknyapun juga harus sama yaitu dalam bentuk DTM). Kalau boleh tahu, df_clean_final yang Ibu lampirkan ini dipakai untuk apa ya bu?
